executor/aggfuncs/func_group_concat.go

```diff
@@ -175,24 +179,23 @@ func (e *groupConcatDistinct) UpdatePartialResult(sctx sessionctx.Context, rowsI
 			if isNull {
 				break
 			}
+			p.encodeBytesBuffer = codec.EncodeBytes(p.encodeBytesBuffer, hack.Slice(v))
 			p.valsBuf.WriteString(v)
 		}
 		if isNull {
 			continue
 		}
-		joinedVals := p.valsBuf.String()
-		if p.valSet.Exist(joinedVals) {
+		if p.valSet.InsertIfNotExist(string(p.encodeBytesBuffer)) {
```

Comment:
Why introduce this new function `InsertIfNotExist`?
How about just replace `joinedVals` with `string(encodeBytesBuffer)` when calling `Exist` and `Insert`.